### PR TITLE
Add basic health check to HDFS

### PR DESCRIPTION
**Pull request by elingg**
This was tested @jsancio or @ConnorDoyle can you take a look please, thx!

### CODE REVIEW

File: repo/packages/H/hdfs/0/marathon.json [hunk 1]
```diff
@@ -5,5 +5,15 @@
   "disk": 100,
   "instances": 1,
   "cmd": "{{#hdfs/custom-config}}echo {{hdfs/custom-config}} | base64 --decode > mesos-site.xml && {{/hdfs/custom-config}}cd hdfs-mesos* && if [ -e '../mesos-site.xml' ]; then cp -f ../mesos-site.xml ./etc/hadoop/; fi && ./bin/hdfs-mesos",
-  "uris": {{{hdfs/scheduler-uris}}}
+  "uris": {{{hdfs/scheduler-uris}}},
+  "healthChecks": [
+    {
+      "protocol": "COMMAND",
+      "command": { "value": "curl -f -X GET http://namenode2.hdfs.mesos:50070" },
```
Comment (ConnorDoyle) [inline repo/packages/H/hdfs/0/marathon.json]: Why namenode 2?

**Comment by BenWhitehead** [comment]
:-1: Marathon shouldn't be responsible for checking the health of tasks of another framework.

**Comment by elingg** [comment]
@benwhitehead, we're looking for a quick working solution before creating a full health check. That said, we can run the hdfs commands instead which is the current plan.

**Comment by elingg** [comment]
What are your thoughts @benwhitehead, @connordoyle? this is a test of listing reading and writing, but we could also do a simple ls command.

**Comment by BenWhitehead** [comment]
This seems like a much better test than the curl one from before.

Have you verified that this sequence of commands can succeed on a cluster that is running?

**Comment by elingg** [comment]
@benwhitehead, yes verified!

File: repo/packages/H/hdfs/0/marathon.json [hunk 2]
```diff
@@ -5,5 +5,15 @@
   "disk": 100,
   "instances": 1,
   "cmd": "{{#hdfs/custom-config}}echo {{hdfs/custom-config}} | base64 --decode > mesos-site.xml && {{/hdfs/custom-config}}cd hdfs-mesos* && if [ -e '../mesos-site.xml' ]; then cp -f ../mesos-site.xml ./etc/hadoop/; fi && ./bin/hdfs-mesos",
-  "uris": {{{hdfs/scheduler-uris}}}
+  "uris": {{{hdfs/scheduler-uris}}},
+  "healthChecks": [
+    {
+      "protocol": "COMMAND",
+      "command": { "value": "hadoop fs -ls hdfs://hdfs/ && hadoop fs -rm -f hdfs://hdfs/test1.txt && rm -f test2.txt && echo \"this is a test\" > test1.txt && hadoop fs -put test1.txt hdfs://hdfs/ && hadoop fs -get hdfs://hdfs/test1.txt test2.txt" },
```
Comment (jsancio) [inline repo/packages/H/hdfs/0/marathon.json]: Looks like you are left with test2.txt in the sandbox.

Also, can we at least make the path for the test configurable and document this behaviour? The last thing we want to do is remove our customer's file.

**Comment by elingg** [comment]
read @jsancio's comment. Perhaps to be safe, we should run a simple ls command. That will fail if hfds is not up so that's a reasonable test. Any objections?

**Comment by ConnorDoyle** [comment]
Nice result :+1:

**Pull request closed without merging**